user/tests/wiring/no_fixture/i2c.cpp

```diff
@@ -115,4 +331,60 @@ test(I2C_06_I2c_Sleep_FuelGauge) {
     auto ver2 = fuel.getVersion();
     assertEqual(ver, ver2);
 }
+
+test(I2C_07_bus_reset_is_not_destructive) {
+    constexpr uint16_t MAX17043_DEFAULT_CONFIG = 0x971c;
+
+    FuelGauge fuel;
+    fuel.begin();
+    fuel.wakeup();
+    auto ver = fuel.getVersion();
+#if HAL_PLATFORM_POWER_MANAGEMENT_OPTIONAL
+    if (ver < 0) {
+        skip();
+        return;
+    }
+#endif // HAL_PLATFORM_POWER_MANAGEMENT_OPTIONAL
+    assertMoreOrEqual(ver, 0);
+    assertNotEqual(ver, 0x0000);
+    assertNotEqual(ver, 0xffff);
+
+    SCOPE_GUARD({
+        fuel.begin();
+        hal_i2c_reset(HAL_PLATFORM_FUELGAUGE_MAX17043_I2C, 0, nullptr);
+        fuel.reset();
+    });
+
+    auto pins = i2cToSdaSclPins(HAL_PLATFORM_FUELGAUGE_MAX17043_I2C);
+    SoftWire wire(HAL_PLATFORM_FUELGAUGE_MAX17043_I2C, pins.first, pins.second);
+
+    hal_i2c_reset(HAL_PLATFORM_FUELGAUGE_MAX17043_I2C, 0, nullptr);
+    fuel.reset();
+
+    // Validate that fuel gauge has default configuration
+    uint8_t buf[2] = {0xff, 0xff};
+    assertEqual(0, fuel.readConfigRegister(buf[0], buf[1]));
+    uint16_t config = buf[0] << 8 | buf[1];
+    assertEqual(config, MAX17043_DEFAULT_CONFIG);
+
+    // Validate that we can read exactly the same configuration with soft i2c implementation
```

Comment:
Could you help me understand why we need to use software I2C to validate the communication with the fuel gauge?

Reply:
We use software I2C to create a situation of an interrupted transaction at a particular location e.g. clock 3 bits out 8 or clock 1 byte and 5 bits and then check whether `hal_i2c_reset` can reliably terminate this ongoing transaction without writing garbage data into the slave (previously it would have just clocked 9 high bits).